src/Testing.NSubstitute/AutoSubstitute.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using DryIoc;
+
+namespace Rocket.Surgery.Extensions.Testing
+{
+    /// <summary>
+    /// Automatically creates substitute for requested services that haven't been registered
+    /// </summary>
+    public class AutoSubstitute : IDisposable
+    {
+        /// <summary>
+        /// Create a container that automatically substitutes unknown types.
+        /// </summary>
+        /// <param name="container"></param>
+        /// <param name="configureAction"></param>
+        public AutoSubstitute(
+            IContainer container = null,
+            Func<IContainer, IContainer>? configureAction = null)
+        {
+            Container = container ?? new Container();
+
+            if (configureAction != null)
+            {
+                Container = configureAction.Invoke(Container);
+            }
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IContainer"/> that handles the component resolution.
+        /// </summary>
+        public IContainer Container { get; }
+
+        /// <summary>
+        /// Resolve the specified type in the container (register it if needed).
+        /// </summary>
+        /// <typeparam name="T">The type of the service.</typeparam>
+        /// <returns>The service.</returns>
+        public T Resolve<T>() => Container.Resolve<T>();
+
+        /// <summary>
+        /// Resolve the specified type in the container (register specified instance if needed).
+        /// </summary>
+        /// <typeparam name="TService">The type of the service.</typeparam>
+        /// <param name="instance">The instance to register if needed.</param>
+        /// <returns>The instance resolved from container.</returns>
+        [SuppressMessage(
+            "Microsoft.Reliability",
+            "CA2000:Dispose objects before losing scope",
+            Justification = "The component registry is responsible for registration disposal."
+        )]
+        public TService Provide<TService>(TService instance)
```

Comment:
There is also a `public TService Provide<TService, TImplementation>()` as a shorthand for registering an implemenation